src/components/side-menu/side-menu.component.ts

```diff
@@ -1,33 +1,56 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, Renderer, animate, state, style, transition, trigger } from '@angular/core';
 
 import { Keymap } from '../../config-serializer/config-items/Keymap';
 import { Macro } from '../../config-serializer/config-items/Macro';
 
 import { UhkConfigurationService } from '../../services/uhk-configuration.service';
 
 @Component({
+    animations: [
+        trigger('toggler', [
+            state('inactive', style({
+                height: '0px'
+            })),
+            state('active',   style({
+                height: '*'
+            })),
+            transition('inactive <=> active', animate('500ms ease-out'))
+        ])
+    ],
     selector: 'side-menu',
     template: require('./side-menu.component.html'),
     styles: [require('./side-menu.component.scss')]
 })
 export class SideMenuComponent implements OnInit {
     private keymaps: Keymap[];
     private macros: Macro[];
+    private animation: {[key: string]: 'active' | 'inactive'};
 
-    constructor(private uhkConfigurationService: UhkConfigurationService) { }
+    constructor(private uhkConfigurationService: UhkConfigurationService, private renderer: Renderer) {
+        this.animation = {
+            keymap: 'active',
+            macro: 'active',
+            addon: 'active'
+        };
+    }
 
     ngOnInit() {
         this.keymaps = this.uhkConfigurationService.getUhkConfiguration().keymaps.elements;
         this.macros = this.uhkConfigurationService.getUhkConfiguration().macros.elements;
     }
 
-    /* tslint:disable:no-unused-variable: This function is used in the template */
-    private toggleHide(event: Event, view: Element) {
-        /* tslint:enable:no-unused-variable */
+    toggleHide(event: Event, type: string) {
```

Comment:
In all examples on official [angular pages](https://angular.io/resources/live-examples/ngmodule/ts/plnkr.html) they use public for functions that are used in templates. Also this way you don't have a problem with tslint. 

I used only event, because if I would use event.target I would need to write it down in three different places in template (and if we add more menus, this number will be bigger). This way I can just write event.target in only one place.